web/modules/custom/ai_screening_project/src/Helper/ProjectHelper.php

```diff
@@ -252,4 +284,102 @@ public function log($level, string|\Stringable $message, array $context = []): v
     $this->logger->log($level, $message, $context);
   }
 
+  /**
+   * Add project group.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The project getting the project tracks added.
+   */
+  private function addProjectGroup(NodeInterface $entity): void {
+    try {
+      // Create group when a project is created.
+      /** @var \Drupal\group\Entity\Group $group */
+      $group = $this->groupStorage->create(['type' => 'project_group']);
+      $group->set('label', 'Group: ' . $entity->label());
+      $group->setOwner($this->userStorage->load($this->accountProxy->id()));
+      $group->save();
+      $group->addRelationship($entity, 'group_node:project');
+      $group->save();
+    }
+    catch (\Exception $exception) {
+      $this->error('Error creating groups: @message', [
+        '@message' => $exception->getMessage(),
+        'entity' => $entity,
+      ]);
+
+      try {
+        /** @var \Drupal\node\Entity\Node $entity */
+        $entity->set(self::FIELD_CORRUPTED, TRUE);
+        $entity->save();
+      }
+      catch (\Throwable) {
+        // Ignore any errors when marking entity as corrupted.
+      }
+    }
+  }
+
+  /**
+   * Add project tracks.
+   *
+   * @param \Drupal\node\NodeInterface $entity
+   *   The project getting the project tracks added.
+   */
+  private function addProjectTracks(NodeInterface $entity): void {
+    try {
+      $projectTrackTermIds = $this->termStorage->getQuery()
+        ->accessCheck(FALSE)
+        ->condition('vid', self::BUNDLE_TERM_PROJECT_TRACK, '=')
+        ->exists('field_webform')
+        ->execute();
+
+      // Add project tracks to project.
+      $projectTrackTerms = $this->termStorage->loadMultiple($projectTrackTermIds);
+
+      foreach ($projectTrackTerms as $projectTrackTerm) {
```

Comment:
It may be worth adding a (simple) drawing showing the relationship between entities in the project (https://mermaid.js.org/syntax/classDiagram.html).

Reply:
Good point